### ISSUE TITLE
fix artifacts resolving as -plain

### ISSUE THREAD

**Issue by codeconsole** [grails/grails-core#13743]
Artifacts should *not* resolve as `-plain` unless a Spring boot version is also resolved.  It is not proper convention to add -plain classifiers to artifacts.

This could result in unwanted dependency resolution when gradle module files are also distributed.

```
   136374  10-11-2024 20:07   BOOT-INF/lib/asset-pipeline-grails-4.5.1-plain.jar
   330271  10-10-2024 05:31   BOOT-INF/lib/gsp-6.2.2-plain.jar
    46996  10-10-2024 05:31   BOOT-INF/lib/sitemesh2-6.2.2-plain.jar
    58173  10-11-2024 20:07   BOOT-INF/lib/hibernate5-8.1.0-plain.jar
   144196  10-10-2024 05:33   BOOT-INF/lib/fields-5.1.0-plain.jar
```

**Comment by matrei** [comment]
grails/grails-gsp#525

**Comment by matrei** [comment]
gpc/fields#377

**Comment by matrei** [comment]
grails/gorm-hibernate5#917

**Comment by codeconsole** [comment]
@matrei can this logic be placed into `grails-gradle-plugin` instead so that it always is applied to plugins?

**Comment by matrei** [comment]
Yes, I was thinking we should do that starting at 7.0.x.

**Comment by codeconsole** [comment]
well, it's unwanted unexpected behavior that did not previously exist so probably should be corrected in 6.2.x and merged forward into 7.0.x

**Comment by matrei** [comment]
Oh, ok, in which version did it start occurring?

**Comment by matrei** [comment]
No matter, I'll add the change to grails-gradle-plugin 6.2.x instead 👍

**Comment by codeconsole** [comment]
@matrei I think it actually started with 6

**Comment by codeconsole** [comment]
Looks like it has been happening with hibernate5 plugin since  2021-05-29 13:37
https://repo.maven.apache.org/maven2/org/grails/plugins/hibernate5/

and gsp starting with 5.0.0.RC1 (2021-07-20), but did NOT occur in < 5.0.0.M1 (2021-07-20)
https://repo.maven.apache.org/maven2/org/grails/plugins/gsp/

**Comment by matrei** [comment]
grails/grails-gradle-plugin#347

**Comment by codeconsole** [comment]
@matrei regarding your reverts, how do you want to handle
```
/ disable main class
bootJar {
    mainClass.set('dummy.Application')
}
findMainClass.onlyIf { false }
```
remove after revert?

**Comment by matrei** [comment]
@codeconsole These can be removed again, as `bootJar` is disabled.

> remove after revert?

Yes

**Comment by matrei** [comment]
Closed by grails/grails-gradle-plugin#347

**Comment by jamesfredley** [comment]
Tickets that will remove -plain from released versions

https://github.com/grails/grails-gsp/issues/546 - gsp & sitemesh2
https://github.com/gpc/fields/issues/380
https://github.com/grails/gorm-hibernate5/issues/932
